### PR TITLE
Fix CIs

### PR DESCRIPTION
**Pull request by hmaarrfk**
I spoke with ukoethe by email and he agreed that dropping python 2.7 would be ok.

Closes: https://github.com/ukoethe/vigra/issues/502

### CODE REVIEW

**Comment by hmaarrfk** [comment]
> 100% tests passed, 0 tests failed out of 55

**Comment by hmaarrfk** [comment]
@k-dominik I am trying to keep this to a "CI only change" and I think I'm close.

I'm struggling with windows. If you have a machine handy and want to debug that  would be greatly appreciated.

OSX is close, but for some reason i'm getting a segfault. fun!

**Comment by k-dominik** [comment]
Hi @hmaarrfk,

awesome effort! I'll look into the windows build tomorrow :)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,166 +1,38 @@
...
-      maxParallel: 2
       matrix:
-        py36:
-          PY_VERSION: 3.6
-        py27:
-          PY_VERSION: 2.7
+        py38:
+          PYTHON_VERSION: 3.8
+        py39:
+          PYTHON_VERSION: 3.8
```
Comment (k-dominik) [inline azure-pipelines.yml:13]:
```suggestion
          PYTHON_VERSION: 3.9
```

Comment (hmaarrfk) [reply azure-pipelines.yml]: Thank you. please do not feel offended if i squash this in the end. Part of this exercise is trying to be "careful" in not destroying things.

**Comment by k-dominik** [comment]
Hi @hmaarrfk,

is there a reason for the testing strategy with OSX-> py37, linux-> py38, py39, and windows-> py310?

**Comment by hmaarrfk** [comment]
> is there a reason for the testing strategy with OSX-> py37, linux-> py38, py39, and windows-> py310?

Just trying to get a "spread" of python versions without building the full matrix. 3 x 4 -> 12 builds is a little "too much".

**Comment by hmaarrfk** [comment]
@k-dominik i'm somewhat tempted to skip OSX and merge this as is. what do you think? There seems to be a few more PRs to modernize the code base that would go through more easily once we have linux + windows CIs in place.

**Comment by k-dominik** [comment]
Hello @hmaarrfk,

yes, let's go ahead and merge this. I am always thinking of getting back to vigra and I promise I will ;). Thank you for persisting here!

Cheers
Dominik

**Pull request merged**